pkg/template/resource_test.go

```diff
@@ -320,3 +321,13 @@ func Test_ResolveBinding_error(t *testing.T) {
 		})
 	}
 }
+
+func Test_ApplyUIDToResourceTemplate(t *testing.T) {
+	uid := Uid()
+	rt := json.RawMessage(`{"name": "something-$(uid)"}`)
```

Comment:
Can you please add more test cases? You can keep most of this code the same by using the table driven tests in the rest of this file. It would be great if you could add test cases for:
- no `$(uid)` (a regular string)
- multiple `$(uid)`s in one string